crates/forge/src/lib.rs

```diff
@@ -290,22 +267,21 @@ pub async fn run(
         let compiled_test_crate = Arc::new(compiled_test_crate);
         let runner_config = runner_config.clone();
         let runner_params = runner_params.clone();
-        let cancellation_tokens = cancellation_tokens.clone();
 
         pretty_printing::print_running_tests(
             compiled_test_crate.tests_location,
             compiled_test_crate.test_cases.len(),
         );
 
-        let summary = run_tests_from_crate(
-            compiled_test_crate,
-            runner_config,
-            runner_params,
-            cancellation_tokens,
-        )
-        .await?;
+        let summary =
+            run_tests_from_crate(compiled_test_crate, runner_config, runner_params).await?;
+
+        let interrupted = summary.interrupted;
 
         summaries.push(summary);
+        if interrupted {
+            break;
+        }
```

Comment:
Why was this introduced (an here specifically)?

Reply:
because we create cancellation channel for each test crate, and each test_crate is executed sync, so if we get info that one of test crate was interupted we can stop process all tests

Reply:
Should we add a comment here?